drivers/gpio/gpio_sam0.c

```diff
@@ -47,122 +49,65 @@ static int gpio_sam0_config(struct device *dev, int access_op, u32_t pin,
 {
 	const struct gpio_sam0_config *config = DEV_CFG(dev);
 	PortGroup *regs = config->regs;
-	u32_t mask = 1 << pin;
-	bool is_out = (flags & GPIO_DIR_MASK) == GPIO_DIR_OUT;
-	int pud = flags & GPIO_PUD_MASK;
-	PORT_PINCFG_Type pincfg;
+	PORT_PINCFG_Type pincfg = {
+		.reg = 0,
+	};
 
-	if (access_op != GPIO_ACCESS_BY_PIN) {
+	if ((flags & GPIO_SINGLE_ENDED) != 0) {
 		return -ENOTSUP;
 	}
 
-	/* Builds the configuration and writes it in one go */
-	pincfg.reg = 0;
-	pincfg.bit.INEN = 1;
-
-	/* Direction */
-	if (is_out) {
-		regs->DIRSET.bit.DIRSET = mask;
-	} else {
-		regs->DIRCLR.bit.DIRCLR = mask;
-	}
-
-	/* Pull up / pull down */
-	if (is_out && pud != GPIO_PUD_NORMAL) {
-		return -ENOTSUP;
-	}
-
-	switch (pud) {
-	case GPIO_PUD_NORMAL:
-		break;
-	case GPIO_PUD_PULL_UP:
-		pincfg.bit.PULLEN = 1;
-		regs->OUTSET.reg = mask;
-		break;
-	case GPIO_PUD_PULL_DOWN:
-		pincfg.bit.PULLEN = 1;
-		regs->OUTCLR.reg = mask;
-		break;
-	default:
-		return -ENOTSUP;
+	/* Supports disconnected, input, output, or bidirectional */
+	if ((flags & GPIO_INPUT) != 0) {
+		pincfg.bit.INEN = 1;
 	}
```

Comment:
we need `} else {` here. Or we should use `WRITE_BIT` macro.

Reply:
`pincfg` is zeroed above, so we only need to set this, not clear it.

Reply:
Sorry, I keep forgetting we're not writing to a register but a variable.